From c5b72fd350bbdfd1facd0ddd5085f238c4cf252a Mon Sep 17 00:00:00 2001
From: Marko Macek <Marko.Macek@gmx.net>
Date: Sat, 21 May 2011 13:30:59 +0100
Subject: [PATCH 01/16] DIX: Set backgroundState correctly for root window

When we change the root window's background to None, and we've run with
-wr or -br for a forced solid background, make sure we also change the
background state to BackgroundPixel, so we don't try to lookup either
pScreen->whitePixel or pScreen->blackPixel as a pixmap.

Signed-off-by: Marko Macek <Marko.Macek@gmx.net>
Reviewed-by: Walter Harms <wharms@bfs.de>
Reviewed-by: Daniel Stone <daniel@fooishbar.org>
Signed-off-by: Keith Packard <keithp@keithp.com>
---
 dix/window.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git xorg-server-1.10.99.901/dix/window.c xorg-server-1.10.99.901/dix/window.c
index cd28d05..5defe58 100644
--- xorg-server-1.10.99.901/dix/window.c
+++ xorg-server-1.10.99.901/dix/window.c
@@ -475,6 +475,7 @@ InitRootWindow(WindowPtr pWin)
         pWin->background.pixel = pScreen->whitePixel;
         backFlag |= CWBackPixmap;
     } else {
+        pWin->backgroundState = BackgroundPixel;
 	if (whiteRoot)
             pWin->background.pixel = pScreen->whitePixel;
         else
@@ -973,6 +974,7 @@ SetRootWindowBackground(WindowPtr pWin, ScreenPtr pScreen, Mask *index2)
     else if (party_like_its_1989)
 	MakeRootTile(pWin);
     else {
+        pWin->backgroundState = BackgroundPixel;
 	if (whiteRoot)
 	    pWin->background.pixel = pScreen->whitePixel;
 	else
-- 
1.7.5.2

